frontend/pages/create-project.tsx

```diff
@@ -0,0 +1,221 @@
+import React, {useState, useEffect} from 'react';
+import Link from 'next/link';
+import { PhotoIcon } from '@heroicons/react/24/solid';
+import { useFormik } from 'formik';
+
+interface CreateProjectProps {}
+
+const CreateProject: React.FC<CreateProjectProps> = () => {
+    const maxCharacterLimit = 20;
+    const [materialsList, setMaterialsList] = useState<string[]>([]);
+
+    const formik = useFormik({
+        initialValues: {
+            projectName: '',
+            aboutProject: '',
+            selectedFile: null,
+            listOfParticipant: [] as string[],
+            newList: '',
+            material: [] as string[],
+            newMaterial: '',
+            error: '',
+        },
+        onSubmit: () => {
+            // Handle form submission
+        },
+    });
+
+    const handleAboutProjectChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const inputValue = e.target.value;
+        if (inputValue.length <= maxCharacterLimit) {
+            formik.setFieldValue('aboutProject', inputValue);
+        }
+    };
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const file = e.currentTarget.files?.[0];
+        formik.setFieldValue('selectedFile', file);
+    };
```

Comment:
Don't send the whole file, read the content as base64
A visual feedback would be nice on upload completion, maybe a preview

Reply:
i did a preview and read in base64